components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/dao/test/APIMgtDAOTest.java

```diff
@@ -1226,35 +1226,6 @@ public void testAddAndGetApi() throws Exception{
         apiMgtDAO.deleteAPI(api.getUuid());
     }
 
-    @Test
-    public void testAddAndConvertNullThrottlingTiers() throws APIManagementException {
-
-        //Adding an API with a null THROTTLING_TIER should automatically convert it to Unlimited
-        APIIdentifier apiIdentifier = new APIIdentifier("testAddAndGetApi", "testAddAndGetApi", "1.0.0");
-        API api = new API(apiIdentifier);
-        api.setContext("/testAddAndGetApi");
-        api.setContextTemplate("/testAddAndGetApi/{version}");
-        Set<URITemplate> uriTemplates = new HashSet<URITemplate>();
-        uriTemplates.add(getUriTemplate("/abc", "GET", "Any", "read", null));
-        api.setUriTemplates(uriTemplates);
-        api.setScopes(getScopes());
-        api.setStatus(APIConstants.PUBLISHED);
-        api.setVersionTimestamp(String.valueOf(System.currentTimeMillis()));
-        api.setAsDefaultVersion(true);
-        api.setUUID(UUID.randomUUID().toString());
-        int apiId = apiMgtDAO.addAPI(api, -1234, "testOrg");
-        apiMgtDAO.addURITemplates(apiId, api, -1234);
-        HashMap<String, String> result1 = apiMgtDAO.getURITemplatesPerAPIAsString(api.getUuid());
-        Assert.assertTrue(result1.containsKey("/abc::GET::Any::Unlimited::abcd defgh fff"));
-
-        //Change the inserted throttling tier back to Null and test the convertNullThrottlingTier method
-        updateThrottlingTierToNull();
-        apiMgtDAO.convertNullThrottlingTiers();
-        HashMap<String, String> result2 = apiMgtDAO.getURITemplatesPerAPIAsString(api.getUuid());
-        Assert.assertTrue(result2.containsKey("/abc::GET::Any::Unlimited::abcd defgh fff"));
-        apiMgtDAO.deleteAPI(api.getUuid());
-    }
-
```

Comment:
Why this test is removed? This should be updated instead. i.e. Adding an API with a null THROTTLING_TIER should automatically convert it to something available.

Reply:
Now, APIs with a null THROTTLING_TIER are not automatically convert it to Unlimited ( Read the comment in line 1232 ). "Unlimited" is a hard coded default value. Now the default value is derived. It may or may not be "Unlimited".